src/include/test/unit_tdb_matrix_with_ids.cc

```diff
@@ -0,0 +1,180 @@
+/**
+ * @file   unit_tdb_matrix_with_matrix.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2023 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ */
+
+#include <catch2/catch_all.hpp>
+#include "detail/linalg/matrix_with_ids.h"
+#include "detail/linalg/tdb_io.h"
+#include "detail/linalg/tdb_matrix_with_ids.h"
+#include "test/test_utils.h"
+
+TEST_CASE("tdb_matrix_with_ids: test test", "[tdb_matrix_with_ids]") {
+  REQUIRE(true);
+}
+
+TEMPLATE_TEST_CASE(
+    "tdb_matrix_with_ids: constructors",
+    "[tdb_matrix_with_ids]",
+    float,
+    double,
+    int,
+    char,
+    size_t,
+    uint32_t) {
+  tiledb::Context ctx;
+
+  std::string tmp_matrix_uri = "/tmp/tmp_tdb_matrix";
+  std::string tmp_ids_uri = "/tmp/tmp_tdb_ids_matrix";
+  int offset = 13;
+  size_t Mrows = 200;
+  size_t Ncols = 500;
+  auto X = ColMajorMatrixWithIds<TestType, TestType, size_t>(Mrows, Ncols);
+  fill_and_write_matrix(
+      ctx, X, tmp_matrix_uri, tmp_ids_uri, Mrows, Ncols, offset);
+  CHECK(X.ids()[0] == offset + 0);
+  CHECK(X.ids()[1] == offset + 1);
+  CHECK(X.ids()[10] == offset + 10);
+
+  auto Y = tdbColMajorMatrixWithIds<TestType, TestType>(
+      ctx, tmp_matrix_uri, tmp_ids_uri);
+  Y.load();
+  CHECK(num_vectors(Y) == num_vectors(X));
+  CHECK(dimension(Y) == dimension(X));
+  CHECK(
```

Comment:
We should also check `CHECK(size(Y.ids()) == size(X.ids()));` and also `CHECK(size(Y.ids()) == Y.num_ids());`

Reply:
Thanks, added more checks.